tests/test_models.py

```diff
@@ -394,19 +420,45 @@ def test_node_update_fn(self):
         global_attributes = None
 
         node_update_fn = hk.transform(node_update_fn)
-        node_update_fn = hk.without_apply_rng(node_update_fn)
         params = node_update_fn.init(
             init_rng,
             nodes, sent_message, received_message, global_attributes
         )
 
         nodes_updated = node_update_fn.apply(
-            params,
+            params, rng,
             nodes, sent_message, received_message, global_attributes
         )
         np.testing.assert_allclose(
             nodes_updated,
             shifted_softplus(received_message['messages']) + nodes)
 
+    def test_readout_function(self):
+        """Test the output of the readout function against the expected result.
+        """
+        rng = jax.random.PRNGKey(42)
+        rng, init_rng = jax.random.split(rng)
+
+        latent_size = 16
+        hk_init = hk.initializers.Identity()
+        readout_node_update_fn = get_readout_node_update_fn(
+            latent_size, hk_init, use_layer_norm=False,
+            activation=shifted_softplus, output_size=1)
+        # TODO: finish test
+
+    def test_mlp(self):
+        """Test the _build_mlp function."""
+        output_sizes = [16, 16, 8]
+        # test with relu activation, and identity weight-matrices
+        '''mlp = hk.transform(
+            _build_mlp(
+                'test_name', output_sizes, use_layer_norm=False,
+                activation=relu, with_bias=False, activate_final=False,
+                w_init=self.hk_init)
+        )'''
+        #inputs = np.ones((1, 8))
+        #outputs = mlp(inputs)
```

Comment:
The output size here looked like a good idea. I imagine haiku has some maybe private properties we can access which give the layer sizes of the network, or number of parameters which would be good to test. Let me know if you're struggling to find access point to those parameters and I'll dig into the hk code as well.